test/e2e/addon/addontest.go

```diff
@@ -0,0 +1,91 @@
+package addon
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go-v2/service/eks"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/go-logr/logr"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+
+	kube "github.com/aws/eks-hybrid/test/e2e/kubernetes"
+)
+
+const (
+	tailLines = 10
+)
+
+type AddonTest struct {
+	clientConfig *rest.Config
+	k8s          kubernetes.Interface
+	eksClient    *eks.Client
+	logger       logr.Logger
+	addon        AddonIface
+}
+
+type AddonIface interface {
+	Setup(ctx context.Context, eksClient *eks.Client, k8s kubernetes.Interface, logger logr.Logger) error
+	CreateAddon(ctx context.Context, eksClient *eks.Client, k8s kubernetes.Interface, logger logr.Logger) error
+	PostInstall(ctx context.Context, eksClient *eks.Client, k8s kubernetes.Interface, logger logr.Logger) error
```

Comment:
is this `PostCreate`? if so, why can't implementation just add that logic in `Create`?